mypy/test/testcheck.py

```diff
@@ -90,15 +91,22 @@ def cases(cls) -> List[DataDrivenTestCase]:
         return c
 
     def run_case(self, testcase: DataDrivenTestCase) -> None:
-        incremental = 'incremental' in testcase.name.lower() or 'incremental' in testcase.file
+        incremental = ('incremental' in testcase.name.lower()
+                       or 'incremental' in testcase.file
+                       or 'serialize' in testcase.file)
         optional = 'optional' in testcase.file
         if incremental:
             # Incremental tests are run once with a cold cache, once with a warm cache.
             # Expect success on first run, errors from testcase.output (if any) on second run.
             # We briefly sleep to make sure file timestamps are distinct.
             self.clear_cache()
-            self.run_case_once(testcase, 1)
-            self.run_case_once(testcase, 2)
+            try:
+                self.run_case_once(testcase, 1)
+                self.run_case_once(testcase, 2)
+            finally:
```

Comment:
There are now three try/finally blocks in this function that resture experiments.STRICT_OPTIONAL. Let's have a single one that does this, wrapping all different cases.